service/controller/resource/statuscondition/create_test.go

```diff
@@ -317,3 +317,370 @@ func newCommonClusterObjectFunc(provider string) func() infrastructurev1alpha2.C
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetCreatingCondition(90),
+				},
+			},
+			nodesReady: false,
+
+			expectedResult: false,
+		},
+		// The cluster already has a created condition
+		{
+			name: "case 2",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetCreatedCondition(60),
+					unittest.GetCreatingCondition(90),
+				},
+			},
+			nodesReady: true,
+
+			expectedResult: false,
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			result := computeCreatedCondition(tc.status, tc.nodesReady)
+			if result != tc.expectedResult {
+				t.Fatalf("expected %v, got %v", tc.expectedResult, result)
+			}
+		})
+	}
+}
+func TestComputeUpdatingCondition(t *testing.T) {
+	testCases := []struct {
+		name string
+
+		status         infrastructurev1alpha2.CommonClusterStatus
+		desiredVersion string
+
+		expectedResult bool
+	}{
+		// the cluster is created and not in updating state but the desired version differs from the current version
+		{
+			name: "case 0",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetCreatedCondition(60),
+					unittest.GetCreatingCondition(90),
+				},
+				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(60, "8.7.5"),
+				},
+			},
+			desiredVersion: "8.7.6",
+
+			expectedResult: true,
+		},
+		// the cluster does not have a created condition
+		{
+			name: "case 1",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetCreatingCondition(90),
+				},
+				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(60, "8.7.5"),
+				},
+			},
+			desiredVersion: "8.7.6",
+
+			expectedResult: false,
+		},
+		// the cluster is already updating
+		{
+			name: "case 2",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetUpdatingCondition(30),
+					unittest.GetCreatedCondition(60),
+					unittest.GetCreatingCondition(90),
+				},
+				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(60, "8.7.5"),
+				},
+			},
+			desiredVersion: "8.7.6",
+
+			expectedResult: false,
+		},
+		// the version is already changed
+		{
+			name: "case 3",
+
+			status: infrastructurev1alpha2.CommonClusterStatus{
+				Conditions: []infrastructurev1alpha2.CommonClusterStatusCondition{
+					unittest.GetCreatedCondition(60),
+					unittest.GetCreatingCondition(90),
+				},
+				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(60, "8.7.6"),
+				},
+			},
+			desiredVersion: "8.7.6",
+
+			expectedResult: false,
+		},
```

Comment:
I would add a case for desired version existing already while the updating condition is also already in place. The result should then be false as well.